### PR TITLE
[clad] Allow building clad as a cling plugin via -DCLING_BUILD_PLUGINS=On

### PR DESCRIPTION
**Pull request by vgvassilev**
cc: @ioanaif

### CODE REVIEW

File: interpreter/cling/tools/plugins/clad/CMakeLists.txt [hunk 1]
```diff
@@ -59,13 +73,9 @@ ExternalProject_Add(
              -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
              -DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}
              -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
-             -DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}
              -DCMAKE_INSTALL_PREFIX=${clad_install_dir}/plugins
-             -DLLVM_DIR=${LLVM_DIR}
-             -DClang_DIR=${Clang_DIR}
```
Comment (oshadura) [inline interpreter/cling/tools/plugins/clad/CMakeLists.txt:65]: @vgvassilev do we need Clang_DIR still?

**Review by oshadura — APPROVED** [review]

**Comment by oshadura** [comment]
@vgvassilev sorry I pushed the wrong button, I want to 'comment' not to 'approve'.

File: interpreter/cling/tools/plugins/clad/CMakeLists.txt (hunk 1 above)
Comment (vgvassilev) [reply interpreter/cling/tools/plugins/clad/CMakeLists.txt:65]: Yes, we need it conditionally. I thought clad improved its automatic detection of this but apparently not...

**Review by oshadura — APPROVED** [review]
LGTM!

**Pull request merged**